rasa/core/lock_store.py

```diff
@@ -329,3 +354,155 @@ def _load_from_module_name_in_endpoint_config(
+        use_ssl: bool = False,
+        key_prefix: Optional[Text] = None,
+        socket_timeout: float = DEFAULT_SOCKET_TIMEOUT_IN_SECONDS,
+    ) -> None:
+        """Create a lock store which uses Redis for persistence.
+
+        Args:
+            host: The host of the redis server.
+            port: The port of the redis server.
+            db: The name of the database within Redis which should be used by Rasa
+                Open Source.
+            password: The password which should be used for authentication with the
+                Redis database.
+            use_ssl: `True` if SSL should be used for the connection to Redis.
+            key_prefix: prefix to prepend to all keys used by the lock store. Must be
+                alphanumeric.
+            socket_timeout: Timeout in seconds after which an exception will be raised
+                in case Redis doesn't respond within `socket_timeout` seconds.
+        """
+        import redis
+
+        self.red = redis.StrictRedis(
+            host=host,
+            port=int(port),
+            db=int(db),
+            password=password,
+            ssl=use_ssl,
+            socket_timeout=socket_timeout,
+        )
+
+        self.key_prefix = DEFAULT_CONCURRENT_REDIS_LOCK_STORE_KEY_PREFIX
+        if key_prefix:
+            logger.debug(f"Setting non-default redis key prefix: '{key_prefix}'.")
+            self._set_key_prefix(key_prefix)
+
+        super().__init__()
+
+    def _set_key_prefix(self, key_prefix: Text) -> None:
+        if isinstance(key_prefix, str) and key_prefix.isalnum():
+            self.key_prefix = (
+                key_prefix + ":" + DEFAULT_CONCURRENT_REDIS_LOCK_STORE_KEY_PREFIX
+            )
+        else:
+            logger.warning(
+                f"Omitting provided non-alphanumeric redis key prefix: '{key_prefix}'. "
+                f"Using default '{self.key_prefix}' instead."
+            )
+
+    def get_lock(self, conversation_id: Text) -> Optional[TicketLock]:
+        """Retrieves lock (see parent docstring for more information)."""
+        tickets: Deque[Ticket] = deque()
+
+        pattern = self.key_prefix + conversation_id + ":" + "[0-9]*"
+        redis_keys = self.red.keys(pattern)
+
+        for key in redis_keys:
+            serialised_ticket = self.red.get(key)
+            ticket = Ticket.from_dict(json.loads(serialised_ticket))
+            tickets.appendleft(ticket)
+
+        tickets = deque(sorted(tickets, key=lambda x: x.number))
+
+        return TicketLock(conversation_id, tickets)
+
+    def delete_lock(self, conversation_id: Text) -> None:
+        """Deletes lock for conversation ID."""
+        pattern = self.key_prefix + conversation_id + ":*"
+        redis_keys = self.red.keys(pattern)
+
+        if not redis_keys:
+            logger.debug(
+                f"The lock store does not contain any key-value "
+                f"items for conversation '{conversation_id}'."
```

Comment:
please add the tried pattern to the debug msg

Reply:
Added: https://github.com/RasaHQ/rasa/pull/11256/commits/016ef68b1e7d325341ca94ca1735a94dc85f36f8